### DISCUSSION TITLE
how to add array to token

### DISCUSSION THREAD

**Discussion by cezarywysocki** [ory/oathkeeper#652] [Q&A] [answered]
when we send data to mutator `hydrator`, it looks like this:
```
{
        "subject": "client8",
        "extra": {
            "aud": [
                "app1",
                "app2"
            ],
            "client_id": "client8",
            "exp": 1614849086,
            "ext": {},
            "iat": 1614762686,
            "iss": "http://ory-hydra.namespace.svc.cluster.local/",
            "jti": "xyz,
            "nbf": 1614762686,
            "scp": [
                "id1",
                "id2",
                "id3"
            ],
            "sub": "client8"
        }
[CUT]
}
```

next we send it to next mutator `id_token`:
```
{
        "handler": "id_token",
        "config": {
          "claims": "{\"aud\": \"{{ print .Extra.aud }}\", \"scope\": \"{{ print .Extra.scp }}\"}"
        }
}
```

but at the end JWT, generated by oathkeeper, looks like:

```
{
  "aud": "[app1 app2]",
  "exp": 1614784756,
  "iat": 1614783856,
  "iss": "http://localhost:4455/",
  "jti": "xyz",
  "nbf": 1614783856,
  "scope": "[id1 id2 id3 id_added_by_hydrator]",
  "sub": "client8"
}
```

how can we fix `aud` and `scope` to looks like below.
I know that we can referrer to indexes like `printIndex .Extra.aud 0` but number of indexes in `aud` and `scope` is not fixed and can be different for each client_id.

```
{
  "aud": "[
      "app1",
      "app2"
   ]",
  "exp": 1614784756,
  "iat": 1614783856,
  "iss": "http://localhost:4455/",
  "jti": "xyz",
  "nbf": 1614783856,
  "scope": "[
      "id1",
      "id2",
      "id3",
      "id_added_by_hydrator"
   ]",
  "sub": "client8"
}

```

**Comment by cezarywysocki** [comment, marked as answer]
as i understood from site gjson.dev [1] `print .Extra.aud` should paste:
`["app1", "app2"]`
but instead it paste:
`[app1 app2]`

so, IMO this is a bug in oathkeeper.

[1]
<img width="652" alt="Screenshot 2021-03-04 at 10 05 23" src="https://user-images.githubusercontent.com/79993838/109940838-671b1c00-7cca-11eb-9287-719ef50a0de2.png">

**Reply by vinckr** [reply]
Thanks for looking into it! 💪 
I checked as well and I think you are correct in that it should format it with `,`

Could you open an issue describing this?

**Reply by cezarywysocki** [reply]
done: https://github.com/ory/oathkeeper/issues/654
